tests/PluginTest.php

```diff
@@ -65,23 +69,81 @@ public function testFileSystem(): void
 
     public function testFunctionExists(): void
     {
-        $plugin = self::loadPlugin("alloc.wasm");
+        $plugin = self::loadPlugin("alloc.wasm", []);
 
         $this->assertTrue($plugin->functionExists("run_test"));
         $this->assertFalse($plugin->functionExists("i_dont_exist"));
     }
 
-    public static function loadPlugin(string $name, ?callable $config = null)
+    public function testHostFunctions(): void
+    {
+        $kvstore = [];
+
+        $kvRead = new HostFunction("kv_read", [ExtismValType::I64], [ExtismValType::I64], function (CurrentPlugin $p, string $key) use (&$kvstore) {
+            return $kvstore[$key] ?? "\0\0\0\0";
+        });
+
+        $kvWrite = new HostFunction("kv_write", [ExtismValType::I64, ExtismValType::I64], [], function (string $key, string $value) use (&$kvstore) {
+            $kvstore[$key] = $value;
+        });
```

Comment:
@mhmd-azeez -- this is pretty neat! the ability to auto-cast (?) these types, e.g. `(string $key, string $value)` in the signature is impressive. Maybe it's just my lack of familiarity with PHP, but I think it would be really helpful to have some notes/docs on this in the source for this callback.

Reply:
Hi, I added parameter details to the README, not sure if that's what you meant. let me know if you think we need more info

Reply:
Those work great - thank you!